go/src/koding/klientctl/endpoint/machine/mount.go

```diff
@@ -0,0 +1,278 @@
+)
+
+// MountOptions stores options for `machine mount` call.
+type MountOptions struct {
+	Identifier string // Machine identifier.
+	Path       string // Machine local path - absolute and cleaned.
+	RemotePath string // Remote machine path - raw format.
+	Log        logging.Logger
+}
+
+// Mount synchronizes directories between remote and local machines.
+func Mount(options *MountOptions) (err error) {
+	if options == nil {
+		return errors.New("invalid nil options")
+	}
+
+	// Create and check mount point directory.
+	clean, err := mountPointDirectory(options.Path)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			clean()
+		}
+	}()
+
+	// TODO(ppknap): this is copied from klientctl old list and will be reworked.
+	k, err := klient.CreateKlientWithDefaultOpts()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error creating klient:", err)
+		return err
+	}
+
+	if err := k.Dial(); err != nil {
+		return err
+	}
+
+	// Translate identifier to machine ID.
+	idReq := machinegroup.IDRequest{
+		Identifier: options.Identifier,
+	}
+	idRaw, err := k.Tell("machine.id", idReq)
+	if err != nil {
+		return err
+	}
+	idRes := machinegroup.IDResponse{}
+	if err := idRaw.Unmarshal(&idRes); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(os.Stdout, "Mounting to %s directory.\nChecking remote path...\n", options.Path)
+
+	m := mount.Mount{
+		Path:       options.Path,
+		RemotePath: options.RemotePath,
+	}
+	// First head the remote machine directory in order to get basic mount info.
+	headMountReq := machinegroup.HeadMountRequest{
+		MountRequest: machinegroup.MountRequest{
+			ID:    idRes.ID,
+			Mount: m,
+		},
+	}
+	headMountRaw, err := k.Tell("machine.mount.head", headMountReq)
+	if err != nil {
+		return err
+	}
+	headMountRes := machinegroup.HeadMountResponse{}
+	if err := headMountRaw.Unmarshal(&headMountRes); err != nil {
+		return err
+	}
+
+	// Remote directory is already mounted to this machine.
+	//
+	// TODO: ask user if she wants to create another mount or stop the process.
+	if headMountRes.ExistMountID != "" {
+		fmt.Fprintf(os.Stdout, "Remote directory %s is already mounted by: %s\n",
+			headMountRes.AbsRemotePath, headMountRes.ExistMountID)
+
+		clean()
+		return nil
+	}
+
+	fmt.Fprintf(os.Stdout, "Mounted remote directory %s has %d file(s) of total size %s\n",
+		headMountRes.AbsRemotePath, headMountRes.AllCount, humanize.IBytes(uint64(headMountRes.AllDiskSize)))
+
+	// TODO: ask user if she wants to continue.
+
+	m.RemotePath = headMountRes.AbsRemotePath
+	fmt.Fprintf(os.Stdout, "Initializing mount %s...\n", m)
+
+	// Create mount.
+	addMountReq := machinegroup.AddMountRequest{
+		MountRequest: machinegroup.MountRequest{
+			ID:    idRes.ID,
+			Mount: m,
+		},
+	}
+	addMountRaw, err := k.Tell("machine.mount.add", addMountReq)
+	if err != nil {
+		return err
+	}
+	addMountRes := machinegroup.AddMountResponse{}
+	if err := addMountRaw.Unmarshal(&addMountRes); err != nil {
+		return err
+	}
```

Comment:
You could add either a method or a helper that would k.Tell + Unmarshal, like [this](https://github.com/koding/koding/blob/master/go/src/koding/klientctl/endpoint/kloud/kloud.go#L25).